src/main/kotlin/com/openlattice/transporter/services/DataTransporterService.kt

```diff
@@ -0,0 +1,202 @@
+
+    /**
+     *  Sync [et] from enterprise to table in atlas
+     */
+    private fun syncTable(et: EntityType): Optional<Future<*>> {
+        val prev = transporterState.putIfAbsent(et.id, TransporterColumnSet(emptyMap()))
+        if (prev != null) {
+            return Optional.empty()
+        }
+        val props = dataModelService.getPropertyTypes(et.properties)
+        return Optional.of(
+                transporterState.submitToKey(et.id, TransporterSynchronizeTableDefinitionEntryProcessor(props).init(data))
+                        .toCompletableFuture()
+        )
+    }
+
+    /**
+     * Get all partitions for [entitySetIds]
+     */
+    private fun partitions(entitySetIds: Set<UUID>): Collection<Int> {
+        return partitionManager.getPartitionsByEntitySetId(entitySetIds).values.flatten().toSet()
+    }
+
+    /**
+     * Regular poll executed by [TransporterRunSyncTask]
+     */
+    fun pollOnce() {
+        val timer = pollTimer.startTimer()
+        val futures = this.transporterState.keys.map { entityTypeId ->
+            val relevantEntitySets = validEntitySets(entityTypeId)
+            val entitySetIds = relevantEntitySets.map { it.id }.toSet()
+            val partitions = partitions(entitySetIds)
+            val ft = transporterState.submitToKey(
+                    entityTypeId,
+                    TransporterPropagateDataEntryProcessor(relevantEntitySets, partitions).init(data)
+            ).toCompletableFuture()
+
+            entitySetIds.count() to ft.get()
+        }
+        val setsPolled = futures.map { it.first }.sum()
+        // wait for all futures to complete.
+        val exception = MultiException()
+        futures.forEach { (_, f) ->
+            try {
+                f
+//                f.get()
+            } catch(e: Exception) {
+                errorCount.inc()
+                exception.add(e)
+            }
+        }
+        val duration = timer.observeDuration()
+        logger.debug("Total poll duration time for {} entity sets in {} entity types: {} sec", setsPolled, futures.size, duration)
+        exception.ifExceptionThrow()
+    }
+
+    private fun refreshDataForEntitySet(entitySetId: UUID) {
+        val esCompletion = entitySets.getAsync( entitySetId )
+        val partitions = partitionManager.getEntitySetPartitions( entitySetId )
+        val refreshTimer = refreshTimer.startTimer()
+        esCompletion.thenCompose{ es ->
+            transporterState.submitToKey(es.entityTypeId,
+                    TransporterPropagateDataEntryProcessor(setOf(es), partitions).init(data)
+            )
+        }.whenCompleteAsync { _, throwable ->
+            refreshTimer.observeDuration()
+            if ( throwable != null ){
+                errorCount.inc()
+            }
+        }
+    }
+
+    private fun validEntitySets(entityTypeId: UUID): Set<EntitySet> {
+        return entitySetService.getEntitySetsOfType(entityTypeId)
+                .filter { !it.isLinking }.toSet()
```

Comment:
will this transport all entity sets? (as opposed to ones with the MATERIALIZE flag?)

Reply:
No, the same check exists elsewhere before transport is done. I'll consolidate them